### PR TITLE
Separate anticheat settings

### PR DESCRIPTION
**Pull request by zmv7**
This PR divides old `disable_anticheat` setting to three settings:
* `anticheat_digging`
* `anticheat_interaction`
* `anticheat_movement`

Both of these settings are enabled by default.

The goal of PR is to make anticheat more configurable.
Movement anticheat causes annoyances on high latency where players get position rubber banding while running, and, according to my experience, also increases CPU usage and server's `max_lag`. So some server owners turn `disable_anticheat` option to `true` and make their own "anti-speedhack" mods. Thus, they get a side effect - cheaters can break nodes very fast. Separated anticheat settings allow to disable that movement anticheat and keep node digging check on. Thus, the server gets more safe.

## To do

This PR is a Work in Progress

- [ ] Add a translations for description of each new setting and remove translations of descriptions of `disable_anticheat` setting
- [ ] Something else?

## How to test

Test each setting using a cheat client

### CODE REVIEW

**Comment by SmallJoker** [comment]
How about looking for improvements on the time tolerances of the anticheat checks instead of simply ignoring the problem by disabling the "bad" sections entirely?

**Comment by zmv7** [comment]
> How about looking for improvements on the time tolerances of the anticheat checks instead of simply ignoring the problem by disabling the "bad" sections entirely?
> 
> I get your motivation for these settings but it only works around an issue instead of fixing the root of the problem.

Uh, it's not so easy to really improve checkMovementCheat(). I can increase the speed limit by 1.5 or 2 times, but will it be a good decision? Though cheaty teleport will be countered anyway, it's the most important...

**Comment by zmv7** [comment]
> > How about looking for improvements on the time tolerances of the anticheat checks instead of simply ignoring the problem by disabling the "bad" sections entirely?
> > I get your motivation for these settings but it only works around an issue instead of fixing the root of the problem.
> 
> Uh, it's not so easy to really improve checkMovementCheat(). I can increase the speed limit by 1.5 or 2 times, but will it be a good decision? Though cheaty teleport will be countered anyway, it's the most important...

Something like this. Practically it helps
```diff
@@ -635,7 +635,7 @@ bool PlayerSAO::checkMovementCheat()
 		player_max_walk = MYMAX(speed_crouch, speed_walk);
 	}
 
-	player_max_walk = MYMAX(player_max_walk, override_max_H);
+	player_max_walk = MYMAX(player_max_walk, override_max_H)*2;
 
 	player_max_jump = m_player->movement_speed_jump * m_player->physics_override.jump;
 	// FIXME: Bouncy nodes cause practically unbound increase in Y speed,

```

**Comment by SmallJoker** [comment]
This is the "time tolerance" part that I was talking about:

https://github.com/minetest/minetest/blob/56902745c8265a8187b2c3855722fa549d1caaf7/src/server/player_sao.cpp#L244-L250

The server lag is - after all - just an estimate. It might be possible debug this further by adding artificial sleep calls into `ServerEnvironment::step`. I did yet not dive into this code but I think the issue is that the pool is decremented by the anticheat (packet handler) before it has the chance to fill up with a new peak dtime of the current (environment) server step that is only taken into account in the next one.

File: po/ru/minetest.po [hunk 1]
```diff
@@ -3197,8 +3197,28 @@ msgid "Digging particles"
...
+msgid "Enable digging cheats prevention in multiplayer."
+msgstr "Включить предотвращение использования читов на быстрое копание в мультиплеере."
+
+#: src/settings_translation_file.cpp
+msgid "Enable anticheat for player interaction"
+msgstr "Античит при взаимодействии"
+
+#: src/settings_translation_file.cpp
+msgid "Enable interaction anticheat in multiplayer."
+msgstr "Включить античит при взаимодействии в мультиплеере."
```
Comment (sfan5) [inline po/ru/minetest.po:3213]: Do not update these files in PRs.

**Comment by sfan5** [comment]
Someone should take a look at the bugs of course but I think these settings are useful.

**Comment by grorp** [comment]
Do we have to keep the `disable_anticheat` setting for backwards compat, so that anticheat stays disabled for server owners who have disabled it previously?

**Comment by lhofhansl** [comment]
Since we're talking about jitter, #13093 is relevant.

**Pull request closed without merging**